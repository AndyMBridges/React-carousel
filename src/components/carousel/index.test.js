import React from 'react';
import {shallow, configure} from 'enzyme';
import renderer from 'react-test-renderer';
import Carousel from './index';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

const images = [{image1: 'image1'}, {image2: 'image2'}, {image3: 'image3'},
    {image4: 'image4'}, {image5: 'image5'}];

const CarouselComponent = shallow(<Carousel images={images} />);

describe('Component: Carousel', () => {

    it('Carousel renders without crashing', () => {
        CarouselComponent;
    });

    it('Carousel snapshot', () => {

        const tree = renderer.create(<Carousel images={images} >test</Carousel>).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('Check right click functionality', () => {

        CarouselComponent.instance().rightClick();

        expect(CarouselComponent.state().direction).toBe('right');
    });

    it('Check left click functionality', () => {

        CarouselComponent.instance().leftClick();

        expect(CarouselComponent.state().direction).toBe('left');
    });

});
