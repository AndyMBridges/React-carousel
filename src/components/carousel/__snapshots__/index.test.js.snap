// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Carousel Carousel snapshot 1`] = `
<div
  className="sc-bxivhb MHFwm"
>
  <span
    className="Carousel__container"
  >
    <div
      className="level2 sc-bdVaJa gMeSEE"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <img
        className="sc-bwzfXH igVZrO"
      />
    </div>
    <div
      className="level1 sc-bdVaJa gMeSEE"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <img
        className="sc-bwzfXH igVZrO"
      />
    </div>
    <div
      className="level0 sc-bdVaJa gMeSEE"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <img
        className="sc-bwzfXH igVZrO"
      />
    </div>
    <div
      className="level-1 sc-bdVaJa gMeSEE"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <img
        className="sc-bwzfXH igVZrO"
      />
    </div>
    <div
      className="level-2 sc-bdVaJa gMeSEE"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <img
        className="sc-bwzfXH igVZrO"
      />
    </div>
  </span>
  <button
    className="sc-htpNat eDRANG"
    direction="prev"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex="0"
  >
    Prev
  </button>
  <button
    className="sc-htpNat ieaWvi"
    direction="next"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex="0"
  >
    Next
  </button>
</div>
`;
